### PR TITLE
Feature/numerical range filter

### PR DESCRIPTION
**Pull request by peterhudec**
## Description of change

This PR adds the `RoutedNumericRangeField` component, based on the `RoutedInput` component. It also refactors the `RoutedInputField` component to be based ot the `RoutedInput`.

## Test instructions

You can see the component in action in the Storybook at `/?path=/story/routednumericrangefield--default`.

Note that the appearance of the component's fieldset element and its legend in Storybook are not how the component looks like in the actual application. This is probably due to missing stylesheets in Storybook.

### CODE REVIEW

File: .storybook/utils.jsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+import {Route} from 'react-router-dom'
+
+export const ResetQueryStringButton = () =>
+  <Route>
+    {({history, location: {search, ...location}}) =>
+      <button onClick={() => history.replace(location)}>
+        Reset querystring
+      </button>
+    }
+  </Route>
```
Comment (rachaelcodes) [inline .storybook/utils.jsx:11]: Just a tiny thing, but could we get a blank line at the EOF?

**Review by rachaelcodes — APPROVED** [review]

File: src/client/components/FilterLabel.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import styled from 'styled-components'
+
+export default styled.label({
+  '-webkit-font-smoothing': 'antialiased',
+  fontWeight: 700,
```
Comment (ian-leggett) [inline src/client/components/FilterLabel.js:5]: This font weight can come from `constants`

File: src/client/components/FilterLabel.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+  display: 'block',
+  color: '#0b0c0c',
```
Comment (ian-leggett) [inline src/client/components/FilterLabel.js:7]: This colour too

File: src/client/components/RoutedInput/Filter.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import styled from 'styled-components'
+
+import RoutedInput from '../RoutedInput'
+
+export default styled(RoutedInput)({
+  fontSize: 19,
```
Comment (ian-leggett) [inline src/client/components/RoutedInput/Filter.js:6]: This font size can come from `constants`

File: src/client/components/RoutedInput/Filter.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  lineHeight: '25px',
+  padding: 6,
+  marginTop: 5,
+  border: '2px solid #0b0c0c',
+  appearance: null,
+  display: 'block',
+  boxSizing: 'border-box',
+  width: '100%',
+  '&:focus': {
+    outline: '3px solid #ffdd00',
```
Comment (ian-leggett) [inline src/client/components/RoutedInput/Filter.js:16]: Colour should come from `govuk-colors`

File: src/client/components/RoutedInput/Filter.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import styled from 'styled-components'
+
+import RoutedInput from '../RoutedInput'
+
+export default styled(RoutedInput)({
+  fontSize: 19,
+  lineHeight: '25px',
+  padding: 6,
+  marginTop: 5,
+  border: '2px solid #0b0c0c',
```
Comment (ian-leggett) [inline src/client/components/RoutedInput/Filter.js:10]: Colour should come from `govuk-colors`

File: src/client/components/FilterLabel.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import styled from 'styled-components'
+
+export default styled.label({
+  '-webkit-font-smoothing': 'antialiased',
```
Comment (ian-leggett) [inline src/client/components/FilterLabel.js:4]: I am surprised this isn't set higher up?

Comment (peterhudec) [reply src/client/components/FilterLabel.js:4]: It took me quite a while to find out why the typeahead labels looked different. They are getting the font smoothing from `.govuk-heading-m`, not sure why is the label not inheriting it.

**Review by ian-leggett — APPROVED** [review]

**Pull request merged**